Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m

```diff
@@ -32,17 +31,20 @@ @interface RandomObject : NSObject
 @property(nonatomic, weak) id delegate;
 @end
 @implementation RandomObject
+#if TARGET_OS_IOS || TARGET_OS_TV
 - (void)application:(UIApplication *)application
 didReceiveRemoteNotification:(NSDictionary *)userInfo
 fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
 }
+#endif
 
 #if __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
 - (void)userNotificationCenter:(UNUserNotificationCenter *)center
        willPresentNotification:(UNNotification *)notification
          withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))
 completionHandler {
 }
+#endif // __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
 
 #if TARGET_OS_IOS
 - (void)userNotificationCenter:(UNUserNotificationCenter *)center
```

Comment:
This seems wrong for < __IPHONE_10_0, but maybe we don't need < __IPHONE_10_0 for the tests?

Reply:
This is only for UNNotificationCenter API which is available for iOS 10+. For iOS 10-, it will just ignore these tests.